core/src/test/scala/com/salesforce/op/ModelInsightsTest.scala

```diff
@@ -161,7 +161,7 @@ class ModelInsightsTest extends FlatSpec with PassengerSparkFixtureTest {
     insights.label.rawFeatureName shouldBe Seq(survived.name)
     insights.label.rawFeatureType shouldBe Seq(survived.typeName)
     insights.label.stagesApplied.size shouldBe 1
-    insights.label.sampleSize shouldBe Some(4.0)
+    insights.label.sampleSize shouldBe Some(5.0)
```

Comment:
@leahmcguire why do you think this was affected? same goes for `area under precision-recall | 1.0` below.

Reply:
These shouldn't be effected unless the data read in changed so we need to track down what is happening

Reply:
@tovbinm @leahmcguire Did some debugging to figure out what's happening. The test uses AggregateAvroReader to read the data. Data has multiple rows for passengerId :4. Aggregated Reader aggregates the data for this particular key. Since the order in which the rows are fetched is changing, aggregation is resulting in change in Data. "Description" field is specifically changing to **"this is description stuff"** . Prior to the change, this field was aggregated as **"stuff this is description"**. Let me know if this could be the reason behind this change needed.

Reply:
@himsmittal as the data that you read in has 6 keys after generation:
```
+---+---------------+--------+---------------------------------------------------------------------------------------------------+
|key|     numericMap|survived|age-description-genderPL-height-numericMap-survived-weight_12-stagesApplied_Prediction_000000000027|
+---+---------------+--------+---------------------------------------------------------------------------------------------------+
|  4|  [Male -> 2.0]|   false|                                                                               [probability_1 ->...|
|  6|           null|    true|                                                                               [probability_1 ->...|
|  2|[Female -> 1.0]|    null|                                                                               [probability_1 ->...|
|  5|[Female -> 1.0]|    null|                                                                               [probability_1 ->...|
|  3|  [Male -> 1.0]|    null|                                                                               [probability_1 ->...|
|  1|[Female -> 1.0]|    null|                                                                               [probability_1 ->...|
+---+---------------+--------+---------------------------------------------------------------------------------------------------+
```
I think this may be due to the combination of the change in ordering and the randomness associated with the holdout set.

Can you add this `  modelWithRFF.score(keepIntermediateFeatures = true).show()` to the test at line 122 and make sure your output looks the same as the one posted above?

Reply:
@leahmcguire added modelWithRFF.score(keepIntermediateFeatures = true).show() to ModelInsightsTest (line 122) . Getting the below output. The columns are different. Am I missing something here ?
```
+---+------------------------------------------------+---------------------------------------------------------------------------------------+--------------------------------------------+-------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------+
|key|numericMap_1-stagesApplied_OPVector_000000000018|age-description-genderPL-height-numericMap-weight_9-stagesApplied_OPVector_000000000019|survived_1-stagesApplied_RealNN_00000000001a|age-description-genderPL-height-numericMap-survived-weight_11-stagesApplied_OPVector_00000000001c|age-description-genderPL-height-numericMap-survived-weight_12-stagesApplied_Prediction_000000000027|
+---+------------------------------------------------+---------------------------------------------------------------------------------------+--------------------------------------------+-------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------+
|  4|                                       [2.0,0.0]|                                                                              [2.0,0.0]|                                         0.0|                                                                                            [0.0]|                                                                               [probability_1 ->...|
|  5|                                       [2.0,1.0]|                                                                              [2.0,1.0]|                                         0.0|                                                                                            [1.0]|                                                                               [probability_1 ->...|
|  6|                                       [2.0,1.0]|                                                                              [2.0,1.0]|                                         1.0|                                                                                            [1.0]|                                                                               [probability_1 ->...|
|  2|                                       [2.0,1.0]|                                                                              [2.0,1.0]|                                         0.0|                                                                                            [1.0]|                                                                               [probability_1 ->...|
|  3|                                       [1.0,0.0]|                                                                              [1.0,0.0]|                                         0.0|                                                                                            [0.0]|                                                                               [probability_1 ->...|
|  1|                                       [2.0,1.0]|                                                                              [2.0,1.0]|                                         0.0|                                                                                            [1.0]|                                                                               [probability_1 ->...|
+---+------------------------------------------------+---------------------------------------------------------------------------------------+--------------------------------------------+-------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------+
```

Reply:
I put in the wrong command :-P `modelWithRFF.score(keepRawFeatures = true).show()` is my printout. However this looks right so I think we are good to go. Thanks for checking!

Reply:
Thanks :)